packages/controls/src/testing/test-definition.ts

```diff
@@ -143,3 +147,40 @@ export function testDefinition<Def extends ControlDefinition>(
     })
   })
 }
+
+export function testResolveValue<Def extends ControlDefinition>(
+  definition: Def,
+  values: readonly ValueType<Def>[],
+) {
+  describe(`resolveValue w/ config ${JSON.stringify(definition.config)}`, () => {
+    test.each(values)('resolves v1 data (%s)', (value) => {
+      const data = definition.toData(value)
+      const resolvedValue = definition.resolveValue(data).readStableValue()
+      if (typeof value !== 'object') {
+        expect(resolvedValue).toBe(value)
+      } else {
+        expect(resolvedValue).toMatchSnapshot()
+      }
```

Comment:
The snapshot here is not used in contrast to `toEqual` or `toB`, but as a way to record the expected value w/o it having propagated all the way from the call site of this helper. We could always use a snapshot, but for all basic controls, if the resolved value is a primitive, it should equal to the original value, and spotting/verifying this fact in snapshots is a lot harder. I guess this could use a comment.